src/transactions/transferFeedUtils.ts

```diff
@@ -400,3 +427,52 @@ export function isTransferTransaction(
 ): tx is TransferItemFragment {
   return (tx as TransferItemFragment).address !== undefined
 }
+
+// Note: This hook is tested from src/transactions/feed/TransferFeedItem.test.ts
+function useFiatConnectTransferDisplayInfo({ amount, transactionHash }: TokenTransfer) {
+  const { t } = useTranslation()
+  const tokenInfo = useTokenInfo(amount.tokenAddress)
+  const fcTransferDetails = useSelector(getCachedFiatConnectTransferSelector(transactionHash))
+  const cachedFiatAccountUses = useSelector(cachedFiatAccountUsesSelector)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+  const account = useMemo(
+    () =>
+      cachedFiatAccountUses.find(
+        (account) => account.fiatAccountId === fcTransferDetails?.fiatAccountId
+      ),
+    [cachedFiatAccountUses, fcTransferDetails]
+  )
+  if (!account || !fcTransferDetails) {
```

Comment:
seems like it would be more efficient to short circuit earlier, before going thru all the cached fiat account uses, if `fcTransferDetails` or `fcTransferDetails.fiatAccountId` was null

Reply:
Updated